pkg/autoscaler/stats_scraper.go

```diff
@@ -146,12 +155,13 @@ func (s *ServiceScraper) Scrape() (Stat, error) {
 	sampleSize := populationMeanSampleSize(readyPodsCount)
 	statCh := make(chan Stat, sampleSize)
 	scrapedPods := &sync.Map{}
+	youngPods := &sync.Map{}
```

Comment:
We could only allocate this if the sampleSize is sufficiently large. We can then check for `!= nil` in the `tryScrape` function below.

Reply:
we'll need a microbenchmark :) False prediction erases whole pipeline and some of the cache :)